src/engraving/libmscore/mscore.h

```diff
@@ -82,28 +84,26 @@ static constexpr int MSCVERSION = 400;
 
 enum class HairpinType : signed char;
 
-static constexpr int VOICES = 4;
+static constexpr size_t VOICES = 4;
 
-static constexpr int INVALID_INDEX = -1;
-
-inline constexpr int staff2track(int staffIdx, int voiceIdx = 0)
+inline constexpr TrackIdx staff2track(StaffIdx staffIdx, VoiceIdx voiceIdx = 0)
 {
-    return staffIdx >= 0 ? staffIdx * VOICES + voiceIdx : INVALID_INDEX;
+    return !mu::is_invalid_idx(staffIdx) ? staffIdx * VOICES + voiceIdx : mu::invalid_idx;
```

Comment:
I would suggest adding mu::is_valid_idx to avoid such inverted conditions

Reply:
This will be incorrect. I can check that the index is definitely not valid, but this does not mean that it is valid, the validity of the index depends on the context of its use.

Example 1
```
size_t idx = mu::invalid_id;
```
This index is definitely not valid

Example 2
```
size_t idx = 9;
```
Is this a valid index? - it is impossible to say definitely

Example 3
```
std::vector<int> vec3 = {1, 2, 3};
size_t idx 3 = 9;

int v = vec.at(idx);
```
In this context, the index is definitely not valid

Reply:
Of course right. `mu::might_be_valid_idx()`  :-)